File: Expenses/src/main/kotlin/br/com/money/expenses/enum/TransactionCategory.kt
package br.com.money.expenses.enum

enum class TransactionCategory (val id : Int) {
    FOOD(1),
    TRANSPORT(2),
    ENTERTAINMENT(3),
    UTILITIES(4),
    HEALTHCARE(5),
    EDUCATION(6),
    SHOPPING(7),
    SALARY(8),
    INVESTMENT(9),
    OTHER(10);

    companion object {
        fun fromId(id: Int): TransactionCategory {
            return entries.find { it.id == id } ?:
                throw IllegalArgumentException("Invalid TransactionCategory id: $id")
        }
    }
}

File: expenses-web/src/types/transactionDto/transactionCategory.ts
// @ts-ignore
export enum TransactionCategory {
    SHOPPING = "SHOPPING",
    FOOD = "FOOD",
    TRANSPORT = "TRANSPORT",
    UTILITIES = "UTILITIES",
    SALARY = "SALARY",
    ENTERTAINMENT = "ENTERTAINMENT",
    HEALTHCARE = "HEALTHCARE",
    EDUCATION = "EDUCATION",
    INVESTMENT = "INVESTMENT",
    OTHER = "OTHER"
}

export type TransactionCategoryType = `${TransactionCategory}`;